app/models.py

```diff
@@ -55,96 +57,38 @@ def find_by_id(self, id):
                 return jsonify({"status": 200,
                         "Redflag": redflag})
 
-    
-    def view(self):
-        return self.db
-  
-    def find_by_id(self, id):
-
-        record = [record for record in self.db if record['Id'] == id]
-
-        return record
-    
-    def post_redflag(self):
+    def patch (self, id):
         data = request.get_json(force=True)
-        CreatedBy = data['CreatedBy']
-        IncidenceType = data['IncidenceType']
-        Location = data['Location']
-        Status = data['Status']
-        Comment = data['Comment'] 
-         
-# return self.save(CreatedBy, IncidenceType, Location, Status, Comment)clear
-# return {
+        record = [record for record in redflag_records if record['Id'] == id]
+        if len(record) == 0:
+            response = jsonify({
```

Comment:
Avoid making responses in your models, all responses should be handled in the views as per the MVC model